### PR TITLE
Core - Rename Mesage to Block

### LINKED ISSUE #435: Core - rename `Message` to `Block`
## Description

During an internal discussion with researchers, we have decided to rename `Message` to `Block` in Stardust to reflect the naming as outlined by the recent IOTA 2.0 whitepapers. The naming change is supposed to help people better understand our technology by using similar names as in other DLTs

### PR DESCRIPTION
**Pull request by oopsmonk**
# Description of change

fixes #435, all tests are passed and tested with Hornet 2.0.0-alpha13

Renamed protocol objects and source files 
- Condition Block -> Unlock Condition
- Unlock Block -> Unlock
- Message -> Block

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

tests are passed with Hornet 2.0.0-alpha13

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/client/api/events/sub_blocks_metadata.c [hunk 1]
```diff
@@ -3,20 +3,20 @@
...
 
-#include "client/api/events/sub_messages_metadata.h"
+#include "client/api/events/sub_blocks_metadata.h"
 #include "client/api/json_parser/json_utils.h"
 #include "client/constants.h"
 #include "client/network/mqtt/mqtt.h"
 
-int event_subscribe_msg_metadata(event_client_handle_t client, int *mid, char const msg_id[], int qos) {
-  if (strlen(msg_id) != BIN_TO_HEX_BYTES(IOTA_MESSAGE_ID_BYTES)) {
+int event_subscribe_blk_metadata(event_client_handle_t client, int *mid, char const msg_id[], int qos) {
```
Comment (rebinmannady93) [inline src/client/api/events/sub_blocks_metadata.c]:
```suggestion
int event_subscribe_blk_metadata(event_client_handle_t client, int *mid, char const block_id[], int qos) {
```

File: benchmark/client/send_tagged_data.c [hunk 2]
```diff
@@ -34,19 +34,19 @@ int main() {
...
-    printf("[%s:%d]: Can not send tagged data message!\n", __func__, __LINE__);
+    printf("[%s:%d]: Can not send tagged data block!\n", __func__, __LINE__);
     res_node_info_free(info);
     return -1;
   }
   res_node_info_free(info);
 
-  printf("[%s:%d]: Message successfully send! URL: http://%s:%d%s/messages/0x%s\n", __func__, __LINE__, NODE_HOST,
-         NODE_PORT, NODE_API_PATH, res.u.msg_id);
+  printf("[%s:%d]: Message successfully send! URL: http://%s:%d%s/blocks/0x%s\n", __func__, __LINE__, NODE_HOST,
```
Comment (rebinmannady93) [inline benchmark/client/send_tagged_data.c]:
```suggestion
  printf("[%s:%d]: Tagged Data successfully send! URL: http://%s:%d%s/blocks/0x%s\n", __func__, __LINE__, NODE_HOST,
```

File: benchmark/client/send_tagged_data_max.c [hunk 3]
```diff
@@ -40,21 +40,21 @@ int main() {
...
     free(tag_data);
     res_node_info_free(info);
     return -1;
   }
   free(tag_data);
   res_node_info_free(info);
 
-  printf("[%s:%d]: Message successfully send! URL: http://%s:%d%s/messages/0x%s\n", __func__, __LINE__, NODE_HOST,
-         NODE_PORT, NODE_API_PATH, res.u.msg_id);
+  printf("[%s:%d]: Message successfully send! URL: http://%s:%d%s/blocks/0x%s\n", __func__, __LINE__, NODE_HOST,
```
Comment (rebinmannady93) [inline benchmark/client/send_tagged_data_max.c]:
```suggestion
  printf("[%s:%d]: Tagged Data successfully send! URL: http://%s:%d%s/blocks/0x%s\n", __func__, __LINE__, NODE_HOST,
```

Comment (rebinmannady93) [inline benchmark/client/send_tagged_data_max.c]:
```suggestion
  printf("[%s:%d]: Tagged Data successfully send! URL: http://%s:%d%s/blocks/0x%s\n", __func__, __LINE__, NODE_HOST,
```

File: tests/core/test_block.c [hunk 4]
```diff
@@ -228,121 +229,121 @@ void test_message_with_tx_serialize() {
   char const* const hornet_faucet = "HORNET FAUCET";
   tagged_data_payload_t* tagged_data = tagged_data_new((byte_t*)hornet_faucet, strlen(hornet_faucet) + 1, NULL, 0);
   TEST_ASSERT_NOT_NULL(tagged_data);
-  TEST_ASSERT(tx_essence_add_payload(essence, CORE_MESSAGE_PAYLOAD_TAGGED, tagged_data) == 0);
+  TEST_ASSERT(tx_essence_add_payload(essence, CORE_BLOCK_PAYLOAD_TAGGED, tagged_data) == 0);
 
   // add signature unlock block
   byte_t* signature = create_signature_unlock_block();
```
Comment (rebinmannady93) [inline tests/core/test_block.c]:
```suggestion
  byte_t* signature = create_signature_unlock();
```

File: examples/client/encrypted_tagged_data_block.c [hunk 5]
```diff
@@ -53,8 +53,8 @@ int main(void) {
 
   // send encrypted data
   printf("Sending encrypted data message to the Tangle...\n");
```
Comment (rebinmannady93) [inline examples/client/encrypted_tagged_data_block.c]:
```suggestion
  printf("Sending encrypted data block to the Tangle...\n");
```

File: examples/client/encrypted_tagged_data_block.c [hunk 6]
```diff
@@ -64,45 +64,45 @@ int main(void) {
     return -1;
```
Comment (rebinmannady93) [inline examples/client/encrypted_tagged_data_block.c:64]:
```suggestion
    printf("Sendng block failed!"\n);
    return -1;
```

File: examples/client/encrypted_tagged_data_block.c [hunk 7]
```diff
@@ -64,45 +64,45 @@ int main(void) {
...
-  if (!msg) {
-    printf("Failed to create a response message object!\n");
+  res_block_t *blk = res_block_new();
+  if (!blk) {
+    printf("Failed to create a response block object!\n");
     return -1;
   }
 
-  // fetch message from the Tangle
+  // fetch the message from the Tangle
```
Comment (rebinmannady93) [inline examples/client/encrypted_tagged_data_block.c]:
```suggestion
  // fetch the block from the Tangle
```

File: examples/client/encrypted_tagged_data_block.c [hunk 8]
```diff
@@ -64,45 +64,45 @@ int main(void) {
...
   printf("Fetching message from the Tangle...\n");
```
Comment (rebinmannady93) [inline examples/client/encrypted_tagged_data_block.c]:
```suggestion
  printf("Fetching block from the Tangle...\n");
```

File: examples/client/encrypted_tagged_data_block.c [hunk 9]
```diff
@@ -64,45 +64,45 @@ int main(void) {
...
     printf("Fetching message from a node failed!\n");
-    res_message_free(msg);
+    res_block_free(blk);
     return -1;
   }
 
-  printf("Message successfully fetched.\n");
+  printf("Block successfully fetched.\n");
 
   // check if fetched message is Tagged Data message
```
Comment (rebinmannady93) [inline examples/client/encrypted_tagged_data_block.c]:
```suggestion
  // check if fetched block is Tagged Data block
```

File: examples/client/encrypted_tagged_data_block.c [hunk 10]
```diff
@@ -64,45 +64,45 @@ int main(void) {
...
-  if (msg->u.msg->payload_type != CORE_MESSAGE_PAYLOAD_TAGGED) {
+  if (blk->u.blk->payload_type != CORE_BLOCK_PAYLOAD_TAGGED) {
     printf("Fetched message is not a Tagged Data message!\n");
```
Comment (rebinmannady93) [inline examples/client/encrypted_tagged_data_block.c]:
```suggestion
    printf("Fetched block is not a Tagged Data block!\n");
```

**Comment by matjazv** [comment]
In `mqtt.h` and `node_event.h` there is `int msg_id;` which needs to be renamed.

**Comment by oopsmonk** [comment]
> In `mqtt.h` and `node_event.h` there is `int msg_id;` which needs to be renamed.

the `msg_id` is the message identifier of the MQTT protocol which is a 16-bit unsigned integer.

**Comment by matjazv** [comment]
`int milestone_deserialize(cJSON* payload, milestone_payload_t* ms)` in `payloads.c` has same `message` names.

**Comment by matjazv** [comment]
In `get_tips.h` there are some `message` names.

**Comment by oopsmonk** [comment]
> `int milestone_deserialize(cJSON* payload, milestone_payload_t* ms)` in `payloads.c` has same `message` names.

> In get_tips.h there are some message names.

done

**Review by rebinmannady93 — APPROVED** [review]

**Review by matjazv — APPROVED** [review]

**Pull request merged**